src/scripts/gems/fluteEffectRunner.ts

```diff
@@ -0,0 +1,126 @@
+                    this.removeEffect(itemName);
+                    Notifier.notify({
+                        message: `The ${GameConstants.humanifyString(itemName)}'s effect ran out!`,
+                        type: NotificationConstants.NotificationOption.danger,
+                        sound: NotificationConstants.NotificationSound.battle_item_timer,
+                        setting: NotificationConstants.NotificationSetting.Items.battle_item_timer,
+                    });
+                }
+            }
+        }
+        this.gemCost();
+    }
+
+    public static getLowestGem(itemName: string) {
+        const item = (ItemList[itemName] as FluteItem);
+        const gemArray = item.gemTypes.map(idx => App.game.gems.gemWallet[PokemonType[idx]]());
+        const gemMaxTime = Math.min(...gemArray);
+        return gemMaxTime;
+    }
+
+    public static updateActiveGemTypes() {
+        this.activeGemTypes.removeAll();
+        let gemNames = [];
+        for (const itemName in GameConstants.FluteItemType) {
+            if (fluteEffectRunner.isActive(GameConstants.FluteItemType[itemName])()) {
+                const item = (ItemList[itemName] as FluteItem);
+                item.gemTypes.forEach(idx => gemNames.push(PokemonType[idx]));
+            }
+        }
+        gemNames = [...new Set(gemNames)];
+        gemNames.forEach(x => this.activeGemTypes.push(x));
+    }
+
+    public static gemCost() {
+        this.activeGemTypes().forEach(idx => App.game.gems.gainGems(-this.numActiveFlutes(), idx));
+    }
+
+    public static getEffect(itemName: string) {
+        if (!player) {
+            return 0;
+        }
+        return player.effectList[itemName]();
+    }
+
+    public static toggleEffect(itemName: string) {
+
+        //NOTE: this if statement is untested
+        if (fluteEffectRunner.getLowestGem(itemName) == 0) {
+            Notifier.notify({
+                message: 'You don\'t have enough gems to use this Flute',
+                type: NotificationConstants.NotificationOption.danger,
+            });
+            return;
+        }
+        if (this.isActive(GameConstants.FluteItemType[itemName])()) {
+            this.removeEffect(itemName);
+            return;
+        }
+
+        player.effectList[itemName](Math.max(0, player.effectList[itemName]() + fluteEffectRunner.getLowestGem(itemName)));
+        GameHelper.incrementObservable(this.numActiveFlutes,1);
+        this.updateFormattedTimeLeft(itemName);
+        this.updateActiveGemTypes();
+    }
+
+    public static removeEffect(itemName: string) {
+        player.effectList[itemName](0);
+        GameHelper.incrementObservable(this.numActiveFlutes, -1);
+        this.updateFormattedTimeLeft(itemName);
+        player.gainItem(itemName, 1);
+        this.updateActiveGemTypes();
+    }
+
+    public static updateFormattedTimeLeft(itemName: string) {
+        const times = GameConstants.formatTime(player.effectList[itemName]()).split(':');
```

Comment:
This will make the time left display show the correct time remaining when multiple flutes are active, you will need the same change for the tooltip (maybe extract this formatTime part into a function which both places can call)
```suggestion
        const times = GameConstants.formatTime(player.effectList[itemName]() / this.numActiveFlutes()).split(':');
```

Reply:
Haven't checked if this causes weirdness when numActiveFlutes = 0, hopefully not.